python/tank/authentication/login_dialog.py

```diff
@@ -114,6 +128,34 @@ def __init__(self, is_session_renewal, hostname=None, login=None, fixed_host=Fal
         self.ui._2fa_code.editingFinished.connect(self._strip_whitespaces)
         self.ui.backup_code.editingFinished.connect(self._strip_whitespaces)
 
+        url = self.ui.site.text()
+        if self._check_sso_enabled(url):
+            self._toggle_sso()
+
+    def _check_sso_enabled(self, url):
+        """
+        Check to see if the web site uses sso.
+
+        We want this method to fail as quickly as possible if there are any
+        issues. Failure is not considered critical, thus known exceptions are
+        silently ignored. At the moment this method is only used to make the
+        GUI show/hide some of the input fields.
+
+        :returns: a boolean indicating if SSO has been enabled or not.
+        """
+        try:
+            # Temporary shotgun instance, used only for the purpose of checking
+            # the site infos.
+            info = Shotgun(url, session_token="dummy", connect=False).info()
+            logger.debug("User authentication method for %s: %s" % (url, info["user_authentication_method"]))
+            if "user_authentication_method" in info:
+                return info["user_authentication_method"] == "saml2"
+        except (ServerNotFoundError, ProtocolError, ValueError, socket.error):
```

Comment:
Personally, I find that doing a `except Exception, e:` without later on doing a `raise e` is bad... But I went ahead as you suggested.